src/main/java/at/ac/tuwien/kr/alpha/common/atoms/ExternalAtom.java

```diff
@@ -91,29 +92,26 @@ public boolean isInternal() {
 
 	@Override
 	public List<VariableTerm> getBindingVariables() {
-		if (negated) {
-			// Negative literal has no binding variables.
-			return Collections.emptyList();
-		}
-		LinkedList<VariableTerm> bindingVariables = new LinkedList<>();
-		for (Term term : input) {
-			bindingVariables.addAll(term.getOccurringVariables());
-		}
-		bindingVariables.addAll(output);
-		return bindingVariables;
+		// If the external atom is negative, then all variables of input and output are non-binding
+		// and there are no binding variables (like for ordinary atoms).
+		// If the external atom is positive, then variables of output are binding.
+		return negated ? emptyList() : unmodifiableList(output);
 	}
 
 	@Override
 	public List<VariableTerm> getNonBindingVariables() {
-		if (!negated) {
-			// Positive literal has only binding variables.
-			return Collections.emptyList();
-		}
+		// External atoms have their input always non-binding, since they cannot
+		// be queried without some concrete input.
 		LinkedList<VariableTerm> nonbindingVariables = new LinkedList<>();
 		for (Term term : input) {
 			nonbindingVariables.addAll(term.getOccurringVariables());
 		}
-		nonbindingVariables.addAll(output);
+
+		// If the external atom is negative, then all variables of input and output are non-binding.
+		if (negated) {
+			nonbindingVariables.addAll(output);
```

Comment:
Similar as above, the output probably must be a list of `Terms` and scanned for occurring variables.